oracle/config/api.go

```diff
@@ -50,6 +50,50 @@ type APIConfig struct {
 type Endpoint struct {
 	// URL is the URL that is used to fetch data from the API.
 	URL string `json:"url"`
+
+	// Authentication holds all data necessary for an API provider to authenticate with
+	// an endpoint.
+	Authentication Authentication `json:"authentication"`
+}
+
+// ValidateBasic performs basic validation of the API endpoint.
+func (e Endpoint) ValidateBasic() error {
+	if len(e.URL) == 0 {
+		return fmt.Errorf("endpoint url cannot be empty")
+	}
+
+	return e.Authentication.ValidateBasic()
+}
+
+// Authentication holds all data necessary for an API provider to authenticate with an
+// endpoint.
+type Authentication struct {
+	// Enabled is a flag that indicates whether the provider should authenticate with
+	// the endpoint.
+	Enabled bool `json:"enabled"`
```

Comment:
Do we need this or can we just use the auth info if it's present and assume it's un-auth'd if there is no auth info.

Reply:
I can remove